### PR TITLE
Auto-save in new Item Submission form breaks the form

### LINKED ISSUE #835: Auto-save in new Item Submission form breaks the form
1. Start creating new Item in collection "Books"
2. Enter Author manually : "John Doe"
3. Enter Title "QWERTY123"
4. Switch to next field - auto save happens as Title is marked as save-on-change field
5. Author values get cleared out !! 
	BUG
6. Fill in all the mandatory fields in all the sections of create form and lastly fill the author fields again
7. The first section "Describe" is still marked as not complete/with errors
	BUG
8. Deposit button is enabled and can be pressed - press that
9. pop-up message appears about error in Deposit process
10. error response in Network tab says : 

`{"timestamp":"2020-08-13T07:11:41.940+0000","status":422,"error":"Unprocessable Entity","message":"Unprocessable or invalid entity","trace":"org.dspace.app.rest.exception.UnprocessableEntityException: Start workflow failed due to validation error on workspaceitem
	at org.dspace.app.rest.submit.SubmissionService.createWorkflowItem(SubmissionService.java:243)
	at org.dspace.app.rest.repository.WorkflowItemRestRepository.createAndReturn(WorkflowItemRestRepository.java:173)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:774)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:691)
	at org.dspace.app.rest.repository.WorkflowItemRestRepository$$EnhancerBySpringCGLIB$$5dbf8ba7.createAndReturn(<generated>)
	at org.dspace.app.rest.repository.DSpaceRestRepository.createAndReturn(DSpaceRestRepository.java:306)
	at org.dspace.app.rest.repository.DSpaceRestRepository$$FastClassBySpringCGLIB$$615525f9.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:687)
	at org.dspace.app.rest.repository.WorkflowItemRestRepository$$EnhancerBySpringCGLIB$$5dbf8ba7.createAndReturn(<generated>)
	at org.dspace.app.rest.RestResourceController.postUriListInternal(RestResourceController.java:460)
	at org.dspace.app.rest.RestResourceController.postWithUriListContentType(RestResourceController.java:405)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:879)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:793)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.dspace.app.rest.filter.DSpaceRequestContextFilter.doFilter(DSpaceRequestContextFilter.java:45)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.dspace.app.rest.filter.ContentLanguageHeaderResponseFilter.doFilter(ContentLanguageHeaderResponseFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.dspace.utils.servlet.DSpaceWebappServletFilter.doFilter(DSpaceWebappServletFilter.java:84)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:126)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:118)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:158)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.dspace.app.rest.security.AnonymousAdditionalAuthorizationFilter.doFilterInternal(AnonymousAdditionalAuthorizationFilter.java:66)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.dspace.app.rest.security.StatelessAuthenticationFilter.doFilterInternal(StatelessAuthenticationFilter.java:101)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:92)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:92)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:77)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.boot.web.servlet.support.ErrorPageFilter.doFilter(ErrorPageFilter.java:128)
	at org.springframework.boot.web.servlet.support.ErrorPageFilter.access$000(ErrorPageFilter.java:66)
	at org.springframework.boot.web.servlet.support.ErrorPageFilter$1.doFilterInternal(ErrorPageFilter.java:103)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.boot.web.servlet.support.ErrorPageFilter.doFilter(ErrorPageFilter.java:121)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:541)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.valves.CrawlerSessionManagerValve.invoke(CrawlerSessionManagerValve.java:235)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:688)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.ajp.AjpProcessor.service(AjpProcessor.java:415)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1639)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)`

### PR DESCRIPTION
**Pull request by alemarte**
## References

Major issue related to this PR: Fixes #835 
Other involved issues are:
[#948](https://github.com/DSpace/dspace-angular/issues/948)
[#864](https://github.com/DSpace/dspace-angular/issues/864)
[#884](https://github.com/DSpace/dspace-angular/issues/884)
[#882](https://github.com/DSpace/dspace-angular/issues/882)
[#931](https://github.com/DSpace/dspace-angular/issues/931)

Before describing the approach adopted in this PR, it is useful to summarise the problems that it directly or indirectly tries to address.
Having investigated them for a long time, it has been possible to trace the root causes of these problems.

- Problems caused by the template/zero row of repeatable fields
    - (1) Errors cannot be displayed correctly 
      Even if the field is correct from an application point of view, the template row is a real form field and follows the validation rules attached to the repeatable field. For this reason we can see a section-level warning. #931

- Problems caused by the repeatable field without form redrawing
  - (2) By clicking on ADD the repeatable field is artificially patched.
    This approach forces the native flow from the used angular library and doesn't work with many of the models implemented in DSPACE, which would need custom patching logic. #948

- Problems caused by the repeatable field with redraw
  -  (3) The content in the template/zero row is deleted. #864 #884

- Problems caused by lookup functionality
  - (4) The redraw of the form is mandatory.
    The selected value is not added to the form, but a save submission is evoked. Once the data returns from the server, the form is redrawn.

- Problems caused by form redraw
  - (5) The history of the form is deleted, bypassing the native angular logic for displaying errors, which is typically very simple to implement by exploiting the touched and pristine fields of the controls.
  - (6) In addition, fields edited but not saved (like during an autosave) are emptied.

At this moment all the above problems except (5), (6) and (3) (which is a side effect) are mitigated by the redraw which is extremely frequent. The more you reduce the frequency of redraw, the more these problems arise.

But it is clear that redraw in turn causes many side effects and ideally should be avoided as much as possible.

**Current workaround to all this mess**

- Notifications to the user have been reduced to only manual saves (by clicking on the submission's footer buttons)
- The Save button is only displayed as enable if there are unsaved changes.
- The hasMetadataEnrichment check responsible to avoid form redraw has been reintroduced by inserting the metadata submission in the store to limit the redraw. This with the aim of reducing (3).
- Unfortunately, as said, this has highlighted a number of weaknesses linked to (2). Solving these kind of problems is out of the scope of this issue, and it would be a very expensive and not very scalable job because it involves writing code for each model, and not very maintainable because in addition to having a lot of code to mantain it forces the native behavior of the library that  should be able to be upgraded easily.
- We did not consider convenient to proceed with solving (3) by saving the values inserted in the first line, unless before a deep analysis and discussion with the community. The problem is the same, as well as making the store heavier it requires the implementation of specific patches for each model, an approach that is difficult to maintain and not scalable.
- Since frequent or infrequent redraw is a scenario to deal with in any case, the information of touched fields has been added for the correct display of errors. But this is all made in vain by (1). Solving this problem without intervening directly on the repeatable field by rethinking it, becomes extremely difficult, unless you rewrite from scratch the error support already provided by angular (which is one of the most valuable aspects of the framework).

Moreover:
We tried to save the focused state of the section in order to postpone the form redraw which causes the emptying of populated fields and the loss of the focus field (6).
The result is that sometimes the redraw is not evoked at all as the user goes from one field to another without ever leaving the form. So we decided to revert (but still in commit history).

It doesn't solve in any way (1) whose fix, as already stated, is expensive to implement and unmaintainable.

**Ideal solution: remove the template row from the repeatable field control**
Removing the template row from the repeatable fields immediately fix (1), (2), (3).
The added submission's metadata into the store reduces the redraw so that the user experience is fluid most of the time. The redraw remains in case of interaction with lookups.
_Moreover_
Implementing the lookup so that it directly patches the form could be more than welcome.
Leave the redraw as the only strategy in case the server actually enriches the form with additional data.
In that case a notification will be displayed to the user and it would probably be acceptable to lose the information about the touched fields and have a solution more clean and maintainable.

## Checklist

- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [ ] My PR doesn't introduce circular dependencies
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @alemarte!

Everything seems works well, barring the exceptions you mention in the description. 
I do have some inline code comments though.

A note to other reviewers: autosave is disabled by default you need to set `environment.submission.autosave.timer` to something higher than 0. Also note that the unit is _minutes_  not milliseconds like other periods in the environment file

**Review by tdonohue — CHANGES_REQUESTED** [review]
@alemarte : I gave this a review & test as well.  First, the code looks good overall, but I agree with all the feedback that @artlowel gives above (so those areas should be resolved).  I added one general comment inline that I think the `timer` should be converted to milliseconds -- but that could be done in a follow-up PR (not required here, but it looks like a small change to me).

I tested this by running `yarn start` and setting the `timer: 1` (one minute).  It mostly works, but I noticed the following oddities:
1. First, if I click around in the submission form (I think it happened when I clicked into the Abstract field), I occasionally see this error appear in Chrome DevTools. This error doesn't seem to affect the submission form, but it should be fixed if possible:
   ```
   ERROR TypeError: Cannot read property 'includes' of undefined
    at section-form.component.ts:230
    at Array.forEach (<anonymous>)
    at l.hasMetadataEnrichment (section-form.component.ts:229)
    at l.updateForm (section-form.component.ts:294)
    at t._next (section-form.component.ts:352)
    at t.__tryOrUnsub (Subscriber.js.pre-build-optimizer.js:194)
    at t.next (Subscriber.js.pre-build-optimizer.js:132)
    at t._next (Subscriber.js.pre-build-optimizer.js:76)
    at t.next (Subscriber.js.pre-build-optimizer.js:53)
    at t._next (distinctUntilChanged.js.pre-build-optimizer.js:55)
   ```
2. I noticed that when the auto-save triggers, I no longer see a notification that the form was saved (in the upper right).  I'm assuming that was removed on purpose?  I'm OK with it's removal if it caused issues, I just was surprised by it.
3. Finally, the first form section *never shows* the green checkbox **unless I enter a name in the "template row" (first row)**.  Here's how I triggered that:
    * I opened the form,
    * I first entered a name, clicked "Add".
    * Then entered a Title & Date
    * I still see an orange exclamation mark (even though all required fields now have values).
    * If I go back to the Author field and typed something in the "Last Name" of the template row, then it immediately turned into a green checkbox. 
4. Strangely, I'm also occasionally able to generate an invalid PATCH (HTTP 500 response).  I've not yet narrowed down how to reproduce this reliably.  But, it occurred when I tried to reproduce (3) above without a file attached.  In that scenario, I was again able to reproduce (3) but also started getting 500 responses from the backend each time the autosave triggered. The 500 response seems to be caused by the PATCH payload is trying to "add" and "remove" the same author field multiple times in a single payload (literally the payload has "add" of an author name, then a "remove" then a second "remove" and then an "add" of the same name again).  This one may be harder to reproduce though, so I'm uncertain whether it's this PR at fault.

If I turn *off* autosave (set `timer: 0`), then I no longer can reproduce (1), (2) or (4).  However, (3) still seems to exist.

As most of these issues seem related to the Author field (a repeatable field), it's possible they are all related to #971 (and the issues described in this ticket description).  So, I'm OK with this PR moving forward (once other feedback is resolved) since it disables autosave by default.  We could move these other issues to the discussion in #971.

File: src/app/submission/form/footer/submission-form-footer.component.ts [hunk 1]
```diff
@@ -49,6 +49,11 @@ export class SubmissionFormFooterComponent implements OnChanges {
    */
   public submissionIsInvalid: Observable<boolean> = observableOf(true);
 
+  /**
+   * A boolean representing if submission form has unsaved modifications
+   */
+  public hasNotSavedModification: Observable<boolean>;
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/form/footer/submission-form-footer.component.ts]: @artlowel Done

File: src/app/shared/form/form.reducer.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ export interface FormEntry {
   data: any;
   valid: boolean;
   errors: FormError[];
+  additional: any;
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/shared/form/form.reducer.ts]: @artlowel Done

The field 'touched' of type FormTouchedState has been added to the FormEntry interface (and all the related code updated accordingly).

File: src/app/submission/objects/submission-objects.effects.ts [hunk 3]
```diff
@@ -389,10 +406,49 @@ export class SubmissionObjectEffects {
...
+/**
+ * Filter sectionErrors accordingly to this rules:
+ * 1. if notifications are enabled return all errors
+ * 2. if sectionType is different from submission-form return all errors
+ * 3. otherwise return errors only for those fields marked as touched inside the section form
+ * @param sectionForm
+ * @param sectionErrors
+ * @param notify
+ */
+function filterErrors(sectionForm, sectionErrors, sectionType, notify): any {
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/objects/submission-objects.effects.ts]: @artlowel Done

File: src/environments/environment.common.ts [hunk 4]
```diff
@@ -60,9 +60,9 @@ export const environment: GlobalConfig = {
   submission: {
     autosave: {
       // NOTE: which metadata trigger an autosave
-      metadata: ['dc.title', 'dc.identifier.doi', 'dc.identifier.pmid', 'dc.identifier.arxiv'],
+      metadata: [],
       // NOTE: every how many minutes submission is saved automatically
-      timer: 5
+      timer: 0
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/environments/environment.common.ts:65]: @tdonohue timer has been refactored to handle milliseconds. Still investigating on the other feedbacks...

**Comment by alemarte** [comment]
@artlowel @tdonohue I've addressed all your feedbacks. I'll comment all of them in a minute. 
Before doing that I have to warn you about a force push that unfortunately I had to perform. I screw up a bit during a rebase with the latest from main to merge a couple of conflicts, very simple indeed. I don't have anymore my local before merging so I had to rebase the only one I had after rebasing. If you have locally the previous copy of the branch you can check that differences are minimal. With this last commit I cleanup and add a missing test for a method.

File: src/app/submission/sections/sections.service.ts [hunk 5]
```diff
@@ -377,4 +394,30 @@ export class SectionsService {
   public setSectionStatus(submissionId: string, sectionId: string, status: boolean) {
     this.store.dispatch(new SectionStatusChangeAction(submissionId, sectionId, status));
   }
+
+  /**
+   * Compute the list of selectable metadata for the section configuration.
+   * @param formConfig
+   */
+  public computeSectionConfiguredMetadata(formConfig: string | SubmissionFormsModel): string[] {
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/sections/sections.service.ts:402]: @artlowel done

File: src/app/shared/form/form.reducer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/shared/form/form.reducer.ts]: https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/shared/form/form.reducer.ts#L27

File: src/app/submission/form/footer/submission-form-footer.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/form/footer/submission-form-footer.component.ts]: https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/submission/form/footer/submission-form-footer.component.ts#L55

File: src/app/submission/objects/submission-objects.effects.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/objects/submission-objects.effects.ts]: https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/submission/objects/submission-objects.effects.ts#L446

File: src/app/submission/objects/submission-objects.reducer.ts [hunk 6]
```diff
@@ -670,6 +713,22 @@ function updateSectionData(state: SubmissionObjectState, action: UpdateSectionDa
   }
 }
 
+/**
+ * Updates the state of the section metadata only when a new value is provided.
+ * Keep the existent otherwise.
+ * @param newMetadata
+ * @param oldMetadata
+ */
+function reduceSectionMetadata(newMetadata: string[], oldMetadata: string[]) {
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/objects/submission-objects.reducer.ts]: @artlowel done

https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/submission/objects/submission-objects.reducer.ts#L724

File: src/app/submission/sections/form/section-form.component.ts [hunk 7]
```diff
@@ -270,15 +289,19 @@ export class SubmissionSectionformComponent extends SectionModelComponent {
...
-      this.isUpdating = false;
-      this.cdr.detectChanges();
+      if (this.hasMetadataEnrichment(sectionData)) {
+        this.isUpdating = true;
+        this.formModel = null;
+        this.cdr.detectChanges();
+        this.initForm(sectionData);
+        this.checksForErrors(errors);
+        this.isUpdating = false;
+        this.cdr.detectChanges();
```
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/sections/form/section-form.component.ts:314]: I didn't touch this lines, but I tried to remove them in turn and side effects occurs, so they would seem to be required.

File: src/app/submission/sections/sections.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (alemarte) [reply src/app/submission/sections/sections.service.ts:402]: https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/submission/sections/sections.service.spec.ts#L384

File: src/environments/environment.common.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (alemarte) [reply src/environments/environment.common.ts:70]: 1. The section metadata is computed asynchronously, so an extra check sounds reasonable in that case. I added it here
   https://github.com/DSpace/dspace-angular/blob/8e77fac63876393f33eb88b8856b29266b981605/src/app/submission/sections/form/section-form.component.ts#L243
2. The notifications are displayed only on manual savings (accordingly to the[ wrapping up](https://github.com/DSpace/dspace-angular/issues/835#issuecomment-709429248) comment related to the 835)
3. This is the very big side effect of the template row of repeatable fields and is the problem I labeled with (1)
4. Is the problem I labeled with (2). The current implementation fix the 'displayed value', but internally form data structures aren't always consistent with the model they'd expect to be.

**Review by artlowel — APPROVED** [review]
Thanks @alemarte!

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me now, @alemarte !  Your feedback makes sense, and (as you noted) several of my previously reported issues will be resolved in #971 (which is a follow-up ticket to this one).

Therefore, I'm OK with merging this as-is, since it definitely **improves** the behavior of auto-save. Further fixes/improvements will be made in the solution to #971 (see that ticket for more details).

**Pull request merged**